src/content/development/contribution-guide/_index.en.md

```diff
@@ -10,23 +10,30 @@ in order for their contributions to the project to be addressed in a timely mann
 
 ## Project Resources
 
-To be expanded
+Submariner uses [GitHub Projects](https://docs.github.com/en/issues/organizing-your-work-with-project-boards) to manage releases. Read
+“[Tracking progress on your project board](https://docs.github.com/en/issues/organizing-your-work-with-project-boards/tracking-work-with-project-boards)"
+to learn more on how to work with projects.
 
 ### Backlog Board
 
-To be expanded
+[The backlog board](https://github.com/orgs/submariner-io/projects/15) hosts issues or features that are not scheduled
+for the release currently being developed.
 
 ### Current Release Board
 
-To be expanded
+Current release work is tracked in [the 0.13 board](https://github.com/orgs/submariner-io/projects/20).
 
 ### Enhancements Repository
 
-To be expanded
+All enhancement proposals need to be submitted to the [enhancements repository](https://github.com/submariner-io/enhancements). To submit a
+new enhancement proposal, raise an 
+[Enhancement Request issue](https://github.com/submariner-io/enhancements/issues/new?assignees=&labels=enhancement&template=enhancement.md)
+on the repository.
```

Comment:
From this phrasing it sounds like any enhancement to submariner should ive here. While this may be useful for unaware users, we should probably move them (or urge to open) to the specific repositories for small/medium tasks.

I'd also document the different usage we have here, e.g. the "issues" are used to propose and track epics, the "pull requests" are used to discuss design proposals for implementing the epics.

Reply:
I think that should be explained under `issues` and `epics` section.